src/Multisig.vy

```diff
@@ -55,16 +55,19 @@ def exec( v: DynArray[uint256, MAX_OWNERS],
           s: DynArray[uint256, MAX_OWNERS],
           target: address, amount: uint256,
           data: Bytes[2000],
+          expiry: uint256
         ):
     assert len(v) == len(r) \
        and len(r) == len(s) \
        and len(s) == self.threshold, 'err/num_sigs'
 
-    txn_hash: bytes32 = keccak256(_abi_encode(TXTYPE_HASH, target, amount, keccak256(data), self.nonce))
+    txn_hash: bytes32 = keccak256(_abi_encode(TXTYPE_HASH, target, amount, keccak256(data), self.nonce, expiry))
     sum_hash: bytes32 = keccak256(concat(convert('\x19\x01', Bytes[2]), DOMAIN_SEPARATOR, txn_hash))
     msg_hash: bytes32 = self.prefix(sum_hash)
     last:     address = empty(address)
 
+    assert expiry >= block.timestamp or expiry == 0, 'err/expired'
```

Comment:
This is treating expiry as 0 as being "unset" expiration. Another approach would be people just send uint256.max.

Reply:
no reason to add this special case logic if people can set arbitrarily high expiry